pkg/proxy/nftables/proxier.go

```diff
@@ -1871,6 +1908,30 @@ func (proxier *Proxier) syncProxyRules() {
 	}
 }
 
+// epChainUpdated returns true if the EP chain doesn't need to be updated.
+func (proxier *Proxier) epChainUpdated(existingChains, existingAffinitySets sets.Set[string], svcInfo *servicePortInfo, epInfo *endpointInfo) bool {
+	if existingChains == nil || existingAffinitySets == nil {
+		// listing existing objects failed, can't skip updating
+		return false
+	}
+	if _, ok := proxier.staleChains[epInfo.chainName]; ok || proxier.staleChainsCleanupFailed {
+		// chain is stale (or could be stale when staleChainsCleanupFailed is true), it was flushed before, needs update
+		return false
+	}
+	// EP chain can have up to 3 rules:
+	// - loopback masquerade rule
+	//   - includes the endpoint IP
+	// - affinity rule when session affinity is set to ClusterIP
+	//   - includes the affinity set name
+	// - DNAT rule
+	//   - includes the endpoint IP + port
+	// EP chain name includes the endpoint IP + port => loopback and DNAT rules are pre-defined by the chain name.
+	// When session affinity is set to ClusterIP, the affinity set is created for local endpoints.
+	// Therefore, we can check that sessions affinity hasn't changes by checking if the affinity set exists.
+	return existingChains.Has(epInfo.chainName) && (svcInfo.SessionAffinityType() == v1.ServiceAffinityClientIP && existingAffinitySets.Has(epInfo.affinitySetName) ||
+		svcInfo.SessionAffinityType() != v1.ServiceAffinityClientIP && !existingAffinitySets.Has(epInfo.affinitySetName))
```

Comment:
could simplify this if you added a temporary variable

    wantAffinitySet := svcInfo.SessionAffinityType() == v1.ServiceAffinityClientIP
    return existingChains.Has(epInfo.chainName) && (wantAffinitySet == existingAffinitySets.Has(epInfo.affinitySetName))